### PR TITLE
util: Update the GitHub Self-Hosted Runners

### PR DESCRIPTION
**Pull request by BobbyRBruce**
1. All VMs are deployable from a single Vagrantfile (per host machine).
2. Runners within VMs are now ephemeral. They cease to exist after a job is complete. After the VM cleans the workspace and creates a new runner. This will reduce old data, scripts, and images causing space issues on our VMs
3. No more 'vm_manager.sh' script. The standard `vagrant` command to manage the VMs will work.
4. Adds Copyright notices where missing.

### CODE REVIEW

**Comment by BobbyRBruce** [comment]
I've created the following VMs using the changes in this PR:

* loupe-test-for-bobby-builder-1
* loupe-test-for-bobby-builder-2
* loupe-test-for-bobby-runner-1
* loupe-test-for-bobby-runner-2

I think they work but, unfortunately, the best test for  us is to just use them and see how they perform. It shouldn't take too long to see any obvious problems.

**Review by mkjost0 — DISMISSED** [review]

**Comment by BobbyRBruce** [comment]
These runners appear to be causing our compiler tests to fail.


https://github.com/gem5/gem5/actions/runs/6348223508/job/17244466433 fails with:

```
Error: Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under '/home/vagrant/_work/_actions/actions/checkout/v3'. Did you forget to run actions/checkout before running your local action?
```

This one appears to just have timed out: https://github.com/gem5/gem5/actions/runs/6348223508/job/17244463049


This one, https://github.com/gem5/gem5/actions/runs/6348223508/job/17244466433, fails with:

```
Error: An error occurred trying to start process '/usr/bin/tar' with working directory '/home/vagrant/_work/_actions/_temp_93e030a2-ec91-4b7a-8ac9-9e416478beb3/_staging'. No such file or directory
```

All the other failures in https://github.com/gem5/gem5/actions/runs/6348223508 appear to have one of these three issues.

While I can't explain the timeout, the other two look like these runners are cleaning up the "_work" directory too early somehow. The job isn't complete but fils are already being removed and causing problems.

File: util/github-runners-vagrant/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+NUM_RUNNERS=0 # Set this to the desired number of runners.
+PERSONAL_ACCESS_TOKEN="<PERSONAL ACCESS TOKEN>"
+GITHUB_ORG="<GITHUB_ORG>"
+HOSTNAME="<VM NAME>"
```
Comment (powerjg) [inline util/github-runners-vagrant/Vagrantfile:33]: I think this works: `ENV['VM_NAME']` and `ENV['TOKEN']`

**Review by powerjg — APPROVED** [review]

**Review by powerjg — CHANGES_REQUESTED** [review]
Remove the commit that shouldn't be here

**Comment by BobbyRBruce** [comment]
> Remove the commit that shouldn't be here

It has been wiped.

**Review by powerjg — APPROVED** [review]

**Pull request merged**